### PR TITLE
Kor 41 player spells b umerge

### PR DESCRIPTION
**Pull request by AlonLivne**

### CODE REVIEW

File: Assets/Scripts/Commands/CastSpellCommand.cs [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+{
+    None,
+    Spell1,
+    Spell2,
+    Spell3
+}
+
+public static class CastSpellCommand  {
+    public static PlayerSpells Spell;
+    public static GameObject SpellObject;
```
Comment (yoavhops) [inline Assets/Scripts/Commands/CastSpellCommand.cs]: SpellObject can be of type PlayerSpell,

File: Assets/Scripts/Spells/AreaSpells.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+           if (Vector3.Distance(unit.transform.position, ClickPosition) <= Radius)
+            {
+                ans.Add(unit);
+            }
+        }
+
+        return ans;
+    }
+
+    public List<EnemyUnit> GetEnemyUnitsInRadius(Vector3 ClickPosition)
```
Comment (yoavhops) [inline Assets/Scripts/Spells/AreaSpells.cs]: this dupilcation

File: Assets/Scripts/Spells/HealSingle.cs [hunk 3]
```diff
@@ -0,0 +1,26 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class HealSingle : SingleUnitSpells {
+
+    public override void OnUpdate()
+    {
+        FindClosestFriend();
```
Comment (yoavhops) [inline Assets/Scripts/Spells/HealSingle.cs]: I would prefer always if possible try to make your code from independent peaces of logic.

Comment (yoavhops) [reply Assets/Scripts/Spells/HealSingle.cs]: world peace please 
piece*

File: Assets/Scripts/Spells/MassHeal.cs [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+using UnityEngine;
+
+public class MassHeal : AreaSpells
+{
+    public List<PlayerUnit> AffectedUnits;
+    public int SpellPower;
+
+    public override void Execute()
+    {
+        AffectedUnits = GetPlayerUnitsInRadius(transform.position);
```
Comment (yoavhops) [inline Assets/Scripts/Spells/MassHeal.cs]: why not local variable?

File: Assets/Scripts/Spells/SingleUnitSpells.cs [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+using UnityEngine;
+
+public class SingleUnitSpells : PlayerSpells {
+    public NewUnit Target;
+
+    public NewUnit FindClosestFriend()
+    {
+        Target = null;
+        var clickPosition = Utils.GetMouseWorldPosition().Value;
+        var dist = 3f;
```
Comment (yoavhops) [inline Assets/Scripts/Spells/SingleUnitSpells.cs]: make spell configuration file and add to there.

File: Assets/Scripts/Spells/HealSingle.cs (hunk 3 above)
Comment (AlonLivne) [reply Assets/Scripts/Spells/HealSingle.cs]: HAHAHA

**Pull request merged**